### PR TITLE
feat!: add `destroy` method to replace current world

### PR DESCRIPTION
**Pull request by christopher-buss**
This is useful for testing purposes, e.g. when we need a brand new world instance between tests.

This is a breaking change as it requires the registry to no longer be a static const export, as the registry can now change.

If there is a different approach to do this, I am happy to take suggestions.

### CODE REVIEW

File: src/registry.ts [hunk 1]
```diff
@@ -28,7 +28,11 @@ export type FilterPairs<T> = {
 };
 
 const components = new Map<string, Entity>();
-export const registry = new ecs.World();
+
+let registry = new ecs.World();
+export function getRegistry(): ecs.World {
+	return registry;
+}
```
Comment (Fireboltofdeath) [inline src/registry.ts:35]: It should be fine to just use `export let registry`, I don't think the function here is necessary

Comment (christopher-buss) [reply src/registry.ts:35]: The problem is query.ts caches the registry import, so every call to query never used the new updated query.

Comment (Fireboltofdeath) [reply src/registry.ts:35]: Does it? It shouldn't

Comment (christopher-buss) [reply src/registry.ts:35]: My unit tests didn't pass until I changed this, unless I am missing something.

In addition though, I don't know how much I like exporting mutable exports regardless.

**Comment by hautajoki** [comment]
imo a better approach would be adding a function to jecs for clearing the world

**Comment by hautajoki** [comment]
also a dirty but quick solution would be creating a new world and directly overwriting the state of the constant registry with the state of the newly created empty world

**Review by Ukendio — APPROVED** [review]

**Pull request closed without merging**